src/structures/Message.ts

```diff
@@ -1,30 +1,29 @@
 import type { Client } from '#client/Client';
-import { Base } from '#structures/Base';
-import type { Channel } from '#structures/Channel';
-import type { Guild } from '#structures/Guild';
 import { User } from '#structures/User';
 import { log } from '#utils/logger';
-import type { APIMessage } from 'discord-api-types/v10';
+import type { APIMessage, RESTPostAPIChannelMessageJSONBody, Snowflake } from 'discord-api-types/v10';
 import { fetch } from 'undici';
-
+import { Base } from './Base';
+import type { Channel } from './Channel';
+import type { Guild } from './Guild';
 export class Message extends Base {
-	public constructor(
-		public id: string,
-		public content: string,
-		public guild: Guild,
-		public channel: Channel,
-		public author: User,
-		client: Client,
-		public reference?: string
-	) {
+	public id: Snowflake = this.data.id;
+	public content: string = this.data.content;
+	public guild?: Guild = this.data.guild_id ? this.client.guilds.get(this.data.guild_id) : undefined;
+	public channel?: Channel = this.client.channels.get(this.data.channel_id);
+	public author: User = new User(this.data.author.id, this.data.author.username, this.data.author.discriminator, this.data.author.bot || false);
```

Comment:
```suggestion
	public author?: User = !this.data.webhook_id ? new User(this.data.author.id, this.data.author.username, this.data.author.discriminator, this.data.author.bot || false) : undefined
```

I was wrong about author being not-nullable, but for a different reason

Reply:
done in https://github.com/TeamAilurus/ailurus/pull/18/commits/6647c4bb628384e536cdd95898c70b62f0774037

Reply:
Maybe add a isWebhook() function as well?